src/fty_mdns_sd_server.cc

```diff
@@ -325,7 +328,17 @@ s_handle_pipe(fty_mdns_sd_server_t* self, zmsg_t **message_p)
         self->fty_info_command = zmsg_popstr (message);
         zsys_debug("fty-mdns-sd-server: DO-DEFAULT-ANNOUNCE %s",
                 self->fty_info_command);
-        int rv=s_poll_fty_info(self);
+        int rv = -1;
+        int tries = 3;
+        while (tries-- >= 0) {
+            rv=s_poll_fty_info(self);
+            if (rv == 0)
+                break;
+            // Wait 5 seconds before retrying
+            zclock_sleep (5000);
```

Comment:
if tries==0 (last failed tentative) it does not make sense to wait for 5 secondes more

Reply:
sure, part of the dirty that goes along the quick. I've to rework the master version to reach the clean state. Thx ;)

Reply:
master version available as PR #22